### PR TITLE
[HOTFIX] Removed BatchedDataSourceScanExec class and extended directly from FileSourceScanExec

### PR DESCRIPTION
**Pull request by ravipesala**
Problem: 
Since some of the code of BatchedDataSourceScanExec is copied from spark, it is difficult to maintain from version upgrades of spark. Currently we face issues during spark 2.3 upgrade so better remove it.

Solution:
Remove BatchedDataSourceScanExec and extend directly the spark class FileSourceScanExec.

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5308/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6478/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5400/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5346/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6514/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5431/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6525/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5357/

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/strategy/CarbonLateDecodeStrategy.scala [hunk 1]
```diff
@@ -673,4 +673,26 @@ private[sql] class CarbonLateDecodeStrategy extends SparkStrategy {
     supportCodegen && vectorizedReader.toBoolean &&
     cols.forall(_.dataType.isInstanceOf[AtomicType])
   }
+
+  private def createHadoopFSRelation(relation: LogicalRelation) = {
+    val sparkSession = relation.relation.sqlContext.sparkSession
+    relation.catalogTable match {
+      case Some(catalogTable) =>
+        HadoopFsRelation(new CatalogFileIndex(
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/strategy/CarbonLateDecodeStrategy.scala]: move `new CatalogFileIndex(` to next line

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/strategy/CarbonLateDecodeStrategy.scala [hunk 2]
```diff
@@ -673,4 +673,26 @@ private[sql] class CarbonLateDecodeStrategy extends SparkStrategy {
...
+          sparkSession,
+          catalogTable, relation.relation.sizeInBytes),
+          catalogTable.partitionSchema,
+          catalogTable.schema,
+          catalogTable.bucketSpec,
+          new SparkCarbonTableFormat,
+          catalogTable.storage.properties)(sparkSession)
+      case _ =>
+        HadoopFsRelation(new InMemoryFileIndex(sparkSession, Seq.empty, Map.empty, None),
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/strategy/CarbonLateDecodeStrategy.scala]: move `new InMemoryFileIndex(sparkSession, Seq.empty, Map.empty, None)` to next line

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/strategy/CarbonLateDecodeStrategy.scala (hunk 1 above)
Comment (jackylk) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/strategy/CarbonLateDecodeStrategy.scala]: If parameter list is long, it is better to add the parameter name also for better readability.

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/strategy/CarbonDataSourceScan.scala [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.catalyst.{InternalRow, TableIdentifier}
+import org.apache.spark.sql.catalyst.expressions.{Attribute, SortOrder}
+import org.apache.spark.sql.catalyst.plans.physical.Partitioning
+import org.apache.spark.sql.execution.FileSourceScanExec
+import org.apache.spark.sql.execution.datasources.{HadoopFsRelation, LogicalRelation}
+
+/**
+ *  Physical plan node for scanning data.
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/strategy/CarbonDataSourceScan.scala]: Please describe whether both `STORED AS CARBONDATA` `USING` will use this physical plan?

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/strategy/CarbonLateDecodeStrategy.scala (hunk 1 above)
Comment (ravipesala) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/strategy/CarbonLateDecodeStrategy.scala]: Ok, moved down.

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/strategy/CarbonLateDecodeStrategy.scala (hunk 2 above)
Comment (ravipesala) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/strategy/CarbonLateDecodeStrategy.scala]: ok

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/strategy/CarbonDataSourceScan.scala (hunk 3 above)
Comment (ravipesala) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/strategy/CarbonDataSourceScan.scala]: At present we don't have the separate reader for `USING`

Comment (ravipesala) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/strategy/CarbonDataSourceScan.scala]: added

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7107/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5883/

**Comment by jackylk** [comment]
LGTM

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5839/

**Pull request closed without merging**